File: internal/node/node.go
package node

import "github.com/Georgecane/openhost/internal/resource"

// LogicalNode represents a logical execution resource assembled from
// independent participant allocations. Its identity is independent of the
// physical participants currently backing it.
type LogicalNode struct {
	ID string

	// Resources is the canonical resource representation exposed by the
	// logical node. Its Capacity is the aggregate resource and its
	// Allocations describe the physical backing.
	Resources resource.CompositeResource

	Runtime RuntimeSpec
}

type RuntimeSpec struct {
	Name    string
	Version string
}

File: src/node.rs
use crate::resource::CompositeResource;

#[derive(Debug, Clone, PartialEq)]
pub struct LogicalNode {
    pub id: String,
    pub resources: CompositeResource,
    pub runtime: RuntimeSpec,
}

#[derive(Debug, Clone, PartialEq, Eq, Default)]
pub struct RuntimeSpec {
    pub name: String,
    pub version: String,
}
